src/components/Vuetify/util/goTo.js

```diff
@@ -0,0 +1,92 @@
+import { consoleError } from '../../../util/console'
+
+const defaults = {
+  duration: 500,
+  offset: 0,
+  easing: 'easeInOutCubic'
+}
+
+const easingPatterns = {
+  // linear
+  linear: t => t,
+  // accelerating from zero velocity
+  easeInQuad: t => t * t,
+  // decelerating to zero velocity
+  easeOutQuad: t => t * (2 - t),
+  // acceleration until halfway, then deceleration
+  easeInOutQuad: t => (t < 0.5 ? 2 * t * t : -1 + (4 - 2 * t) * t),
+  // accelerating from zero velocity
+  easeInCubic: t => t * t * t,
+  // decelerating to zero velocity
+  easeOutCubic: t => --t * t * t + 1,
+  // acceleration until halfway, then deceleration
+  easeInOutCubic: t => t < 0.5 ? 4 * t * t * t : (t - 1) * (2 * t - 2) * (2 * t - 2) + 1,
+  // accelerating from zero velocity
+  easeInQuart: t => t * t * t * t,
+  // decelerating to zero velocity
+  easeOutQuart: t => 1 - --t * t * t * t,
+  // acceleration until halfway, then deceleration
+  easeInOutQuart: t => (t < 0.5 ? 8 * t * t * t * t : 1 - 8 * --t * t * t * t),
+  // accelerating from zero velocity
+  easeInQuint: t => t * t * t * t * t,
+  // decelerating to zero velocity
+  easeOutQuint: t => 1 + --t * t * t * t * t,
+  // acceleration until halfway, then deceleration
+  easeInOutQuint: t => t < 0.5 ? 16 * t * t * t * t * t : 1 + 16 * --t * t * t * t * t
+}
+
+function getDocumentHeight () {
+  return Math.max(
+    document.body.scrollHeight,
+    document.documentElement.scrollHeight,
+    document.body.offsetHeight,
+    document.documentElement.offsetHeight,
+    document.body.clientHeight,
+    document.documentElement.clientHeight
+  )
+}
+
+function getTargetLocation (target, settings) {
+  const documentHeight = getDocumentHeight()
+  const windowHeight = window.innerHeight || (document.documentElement || document.body).clientHeight
+
+  let location
+
+  if (target instanceof Element) location = target.offsetTop
+  else if (typeof target === 'string') location = document.querySelector(target).offsetTop
+  else if (typeof target === 'number') location = target
+  else location = undefined
+
+  location += settings.offset
+
+  return Math.round(
+    documentHeight - location < windowHeight
+      ? documentHeight - windowHeight
+      : location
+  )
+}
+
+export default function goTo (target, options) {
+  if (typeof window === 'undefined') return
+
+  const settings = Object.assign({}, defaults, options)
+
+  const startTime = performance.now()
+  const startLocation = window.pageYOffset
+  const targetLocation = getTargetLocation(target, settings)
+  const distanceToScroll = targetLocation - startLocation
+
+  if (isNaN(targetLocation)) return consoleError(`Target must be a Selector/Number/DOMElement, received ${target.constructor.name} instead.`)
+
+  function step (currentTime) {
+    let progressPercentage = Math.min(1, ((currentTime - startTime) / settings.duration))
+    let easeFunction = easingPatterns[settings.easing](progressPercentage)
```

Comment:
Would be nice if `easing` could be a user provided function as well. And maybe `easingPatterns` could be extracted to a separate helper so that it could be used in other context. And maybe `easing` could be just a function instead of function/string? That it will also easier to do code autocompletion
```js
import { easeInQuad } from '.../easing'
import goTo from '.../goTo'

goTo(target, {
  easing: easeInQuad
})
```
 
`easeFunction` suggests that it's a function, but it's a number - however `easeFunction` could be a function, just evaluate it outside of the `step()` and then do `distanceToScroll * easeFunction(progressPercentage)`. Of course we allow only functions then there's need to do it, but still the var name should be changed

Reply:
I can add a `customEasing` prop inside the options object. Externalizing the easing functions as you did adds an extra importing step. Isn't that needless complication?

Reply:
Definitely don't want another prop.
And I don't think that one more import is issue, at least the possibility of reusing these functions is more important (of course you could export them in `goTo.js` but importing easing fn from `goTo` doesn't sound right for me)

Reply:
While I think it's a good idea to break out the easing functions to a separate helper, it should be goTo that imports them, not the user. imo `easing´ should take either a string (look up in imported easings) or a function to provide a custom one. I don't see a problem with goTo importing the easings, as it does depend on them to function properly.

Reply:
I agree with neko. If `easing` is a string -> look it up in the patterns, if it's a function use it.

Reply:
Fine for me, but move the evaluation of easing function outside of `step` function

```js
const easingFunction = easing is function ? easing : patterns[easing]

function step() {
  easing = easignFunction(percent)
}
```

Reply:
and `consoleError('Easing function ... not found')` if the function with provided name doesn't exist